drivers/hid/hid-ids.h

```diff
@@ -385,7 +385,7 @@
 #define USB_DEVICE_ID_DWAV_EGALAX_MULTITOUCH_73F7	0x73f7
 #define USB_DEVICE_ID_DWAV_EGALAX_MULTITOUCH_A001	0xa001
 
-#define I2C_PRODUCT_ID_ELAN_TOUCHPAD 0x303e
+#define USB_DEVICE_ID_ELAN_TOUCHPAD 0x303e
```

Comment:
You need to put this line below the "#define USB_VENDOR_ID_ELAN" line (and then sorted by product-id, so below the USB_DEVICE_ID_HP_X2_10_COVER define).